plugins/modules/fabric_virtual_networks_workflow_manager.py

```diff
@@ -0,0 +1,2950 @@
+            collected_add_anycast_payload (dict): A dictionary containing the necessary details for
+                adding Anycast Gateways.
+        Returns:
+            self (object): The instance of the class, allowing for method chaining. The status of the operation
+                can be checked via the `status` attribute.
+        Description:
+            This function interacts with the Cisco DNA Center API to add Anycast Gateways. It sends the provided
+            payload to the API and processes the response.
+            In case of an exception during the API call, the function captures the exception and logs the error.
+            The method returns the instance itself, allowing for further interactions with the object.
+        """
+
+        try:
+            payload = {"payload": collected_add_anycast_payload}
+            task_name = "add_anycast_gateways"
+            task_id = self.get_taskid_post_api_call("sda", task_name, payload)
+
+            if not task_id:
+                self.msg = "Unable to retrive the task_id for the task '{0}'.".format(task_name)
+                self.set_operation_result("failed", False, self.msg, "ERROR")
+                return self
+
+            success_msg = "Anycast Gateway(s) '{0}' added successfully in the Cisco Catalyst Center.".format(self.created_anycast)
+            self.get_task_status_from_tasks_by_id(task_id, task_name, success_msg)
+
+        except Exception as e:
+            self.msg = (
+                "An exception occured while adding the Anycast Gateway(s) '{0}' in the Cisco Catalyst "
+                "Center: {1}"
+            ).format(self.created_anycast, str(e))
+            self.set_operation_result("failed", False, self.msg, "ERROR")
+
+        return self
+
+    def update_anycast_gateways_in_system(self, collected_update_anycast_payload):
+        """
+        Updates Anycast Gateways in the Cisco Catalyst Center using the provided payload.
+        Args:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+            collected_update_anycast_payload (dict): A dictionary containing the necessary details for updating
+                Anycast Gateways.
+        Returns:
+            self (object): The instance of the class, allowing for method chaining. The status of the operation
+                can be checked via the `status` attribute.
+        Description:
+            This function interacts with the Cisco DNA Center API to update Anycast Gateways. It sends the provided
+            payload to the API and processes the response.
+            In case of an exception during the API call, the function captures the exception and logs the error.
+            The method returns the instance itself, allowing for further interactions with the object.
+        """
+
+        try:
+            payload = {"payload": collected_update_anycast_payload}
+            task_name = "update_anycast_gateways"
```

Comment:
self.log("Constructing API call payload for task '{0}': {1}".format(task_name, payload), "DEBUG")